game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleSteps.java

```diff
@@ -51,24 +55,49 @@
   final @NonNull Collection<Unit> defendingUnits;
 
   final @NonNull Collection<Unit> attackingWaitingToDie;
+
+  @Getter(onMethod = @__({@Override}))
   final @NonNull Collection<Unit> defendingWaitingToDie;
+
   final @NonNull Territory battleSite;
+
+  @Getter(onMethod = @__({@Override}))
   final @NonNull GameData gameData;
+
   final @NonNull Collection<Unit> bombardingUnits;
   final @NonNull Function<Collection<Unit>, Collection<Unit>> getDependentUnits;
   final @NonNull Boolean isBattleSiteWater;
+
   final @NonNull Boolean isAmphibious;
   final @NonNull Supplier<Collection<Territory>> getAttackerRetreatTerritories;
   final @NonNull Function<Collection<Unit>, Collection<Territory>> getEmptyOrFriendlySeaNeighbors;
   final @NonNull BattleActions battleActions;
 
+  final @NonNull Boolean isOver;
+
+  @Override
+  public Collection<Territory> getAttackerRetreatTerritories() {
+    return getAttackerRetreatTerritories.get();
+  }
+
+  @Override
+  public boolean isOver() {
```

Comment:
I couldn't use `@Getter` because isOver is a `Boolean` and not a `boolean`.  `@Getter` will create a `is` function only for `boolean`.  I could change isOver to be a `boolean` and then I can't have the `@NonNull`.